tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -201,6 +214,83 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
                 offset += WriteMetaStruct(ref bytes, offset, in spanModel);
             }
 
+            if (span.SpanLinks is { Count: > 0 })
+            {
+                offset += WriteSpanLink(ref bytes, offset, in spanModel);
+            }
+
+            return offset - originalOffset;
+        }
+
+        private int WriteSpanLink(ref byte[] bytes, int offset, in SpanModel spanModel)
+        {
+            int originalOffset = offset;
+            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _spanLinkBytes);
+            offset += MessagePackBinary.WriteArrayHeader(ref bytes, offset, spanModel.Span.SpanLinks.Count);
+            foreach (var spanLink in spanModel.Span.SpanLinks)
+            {
+                var context = spanLink.Context;
+                var traceState = W3CTraceContextPropagator.CreateTraceStateHeader(context);
+                // 3 possible values, 1, 0 or null
```

Comment:
This is my fault because I said this yesterday, but this comment is not technically correct. There are multiple values for sampling priority (`-1`, `0`, `1`, `2`, `null`). What I meant is that there are three cases we care about here: `> 0`, `<= 0`, and `null`. I will suggest a different comment below.
```suggestion
```